packages/block-editor/src/hooks/generated-class-name.js

```diff
@@ -20,6 +20,7 @@ import { hasBlockSupport, getBlockDefaultClassName } from '@wordpress/blocks';
  * @return {Object} Filtered props applied to save element.
  */
 export function addGeneratedClassName( extraProps, blockType ) {
+	if ( blockType.name === 'core/comments' ) return extraProps;
```

Comment:
The old `save` function added to the `deprecated` doesn't work yet because this function (called in the `blocks.getSaveContent.extraProps` filter) is adding the `wp-comments` class name.

Reply:
> called in the `blocks.getSaveContent.extraProps` filter

Can we add our own filter to that hook in order to remove the old class name? 🤔

Reply:
Oooh, or maybe we can solve our issue using [this filter](https://developer.wordpress.org/block-editor/reference-guides/filters/block-filters/#blocks-getblockdefaultclassname)?

Reply:
> > called in the `blocks.getSaveContent.extraProps` filter
> 
> Can we add our own filter to that hook in order to remove the old class name? 🤔

```js
function setCommentsBlockCustomClassName( props ) {
	let { className } = props;
	const classes = className?.split( ' ' );
	if ( classes.includes( 'wp-block-comments' ) ) {
		className = classes
			.filter( ( c ) => c !== 'wp-block-comments-query-loop' )
			.join( ' ' );
	}
	return { ...props, className };
}

addFilter(
	'blocks.getSaveContent.extraProps',
	'core/comments/set-block-custom-class-name',
	setCommentsBlockCustomClassName
);
```

![image](https://user-images.githubusercontent.com/96308/165643315-45bea924-01e4-48f4-afd7-169f149ea154.png)


Well, not quite there yet 😅